### PR TITLE
reduce compiler warnings

### PR DESCRIPTION
**Pull request by shailesh33**
no major changes just added some typecasts etc wherever necessary
tried to change some variable types where appropriate
reduced some logging levels that I accidentally bumped up
Simplified some logic in receiving encrypted data
Took care of some shadow declarations.

Overall the new code is fairly warning free. The old code still has some warnings

### CODE REVIEW

File: src/dyn_core.c [hunk 1]
```diff
@@ -64,7 +64,6 @@ core_print_peer_status(void *arg1)
             }
         }
     }
```
Comment (ipapapa) [inline src/dyn_core.c:66]: why don't we need this task anymore?

File: src/dyn_core.h [hunk 2]
```diff
@@ -294,7 +294,7 @@ struct server_pool {
     int                backlog;              /* listen backlog */
     uint32_t           client_connections;   /* maximum # client connection */
     msec_t             server_retry_timeout_ms; /* server retry timeout in msec */
-    uint32_t           server_failure_limit; /* server failure limit */
+    uint8_t            server_failure_limit; /* server failure limit */
```
Comment (ipapapa) [inline src/dyn_core.h:297]: up to 255 only?

File: src/proto/dyn_redis.c [hunk 3]
```diff
@@ -2579,6 +2579,7 @@ redis_pre_coalesce(struct msg *rsp)
          * MSG_RSP_REDIS_MULTIBULK. For an invalid response, we send out -ERR
          * with EINVAL errno
          */
+        log_info("Invalid Response type");
```
Comment (ipapapa) [inline src/proto/dyn_redis.c]: should this be a warning?

**Review by ipapapa — APPROVED** [review]

File: src/dyn_core.c (hunk 1 above)
Comment (shailesh33) [reply src/dyn_core.c:66]: too much logging. every 30 secs

File: src/dyn_core.h (hunk 2 above)
Comment (shailesh33) [reply src/dyn_core.h:297]: not enough?

**Pull request merged**